src/test/java/org/broadinstitute/hellbender/tools/walkers/coverage/DepthOfCoverageIntegrationTest.java

```diff
@@ -0,0 +1,259 @@
+        cmd += " -O "+output.getAbsolutePath();
+        runCommandLine(cmd.split(" "));
+
+        File[] actualFiles = baseOutputFile.listFiles();
+
+        compareOutputDirectories(expectedBaseName, output.getName(), actualFiles);
+    }
+
+    @Test
+    // NOTE, the gene list file was not generated with GATK3 due to different gene list merging behavior
+    public void testGeneListDataAllCovered() throws IOException {
+        final String expectedBaseName = "testGeneListDataAllCovered";
+        final File baseOutputFile = createTempDir("testNoCoverageDueToFiltering");
+        final File output = IOUtils.createTempFileInDirectory( "testNoCoverageDueToFiltering", ".csv", baseOutputFile);
+        output.delete();
+
+        String[] cmd = new String[]{ "-R",hg38Reference,
+                "-I",largeFileTestDir+"multiSampleSubsetted.bam",
+                "-L",getTestFile("artificial.gene_target.interval_list").getAbsolutePath(),
+                "--calculate-coverage-over-genes",getTestFile("refGene_CDK11B.refseq").getAbsolutePath(),
+                "--min-base-quality","0","-dels","-pt","sample","--output-format","CSV","--omit-depth-output-at-each-base","--omit-per-sample-statistics",
+                "-O",output.getAbsolutePath()};
+        runCommandLine(cmd);
+
+        File[] actualFiles = baseOutputFile.listFiles();
+
+        compareOutputDirectories(expectedBaseName, output.getName(), actualFiles);
+    }
+
+    @Test
+    public void testIntervalListOverlappingUniqueBehavior() throws IOException {
+        final String expectedBaseName = "testIntervalListOverlappingUniqueBehavior"; //interval_summary files were not generated with gatk3 as they behave by interval merging
+        final File baseOutputFile = createTempDir("testNoCoverageDueToFiltering");
+        final File output = IOUtils.createTempFileInDirectory( "testNoCoverageDueToFiltering", ".csv", baseOutputFile);
+        output.delete();
+
+        ArgumentsBuilder cmd = new ArgumentsBuilder();
+        cmd.addReference(new File(hg38Reference));
+        cmd.addInput(new File(largeFileTestDir+"multiSampleSubsetted.bam"));
+        cmd.addArgument("min-base-quality","0");
+        cmd.addArgument("include-deletions");
+        cmd.addArgument("partition-type", "sample");
+        cmd.addArgument("omit-depth-output-at-each-base");
+        cmd.addOutput(output);
+
+        // We construct overlapping intervals to demonstrate the new interval merging behavior
+        for (int i = 1656275; i < 1656975; i+=10) {
+            cmd.addInterval(new SimpleInterval("chr1", i, i+100));
+        }
+        runCommandLine(cmd);
+        File[] actualFiles = baseOutputFile.listFiles();
+
+        compareOutputDirectories(expectedBaseName, output.getName(), actualFiles);
+    }
+
+
+    @Test
+    // NOTE, the gene list file was not generated with GATK3 due to different gene list merging behavior
+    public void testGeneListDataPartallylCovered() throws IOException {
```

Comment:
`testGeneListDataPartallylCovered` -> `testGeneListDataPartiallyCovered`

Reply:
Also add a comment explaining how this test case differs from the `testGeneListDataAllCovered` test case above